### PR TITLE
Deprecate core/view/homeDirectory API endpoint

### PR DESCRIPTION
**Pull request by ricekot**
Also change the default value of OptionsParam#userDirectory from null
to the user's home directory.

Signed-off-by: ricekot <ricekot@gmail.com>

### CODE REVIEW

File: zap/src/main/java/org/parosproxy/paros/model/OptionsParam.java [hunk 1]
```diff
@@ -92,7 +93,8 @@
     private Vector<AbstractParam> paramSetList = new Vector<>();
     private Map<Class<? extends AbstractParam>, AbstractParam> abstractParamsMap = new HashMap<>();
     private boolean gui = true;
-    private File userDirectory = null;
+    // ZAP: Change the default value of userDirectory from null.
```
Comment (kingthorin) [inline zap/src/main/java/org/parosproxy/paros/model/OptionsParam.java:96]: Move this to L44/45?

Comment (ricekot) [reply zap/src/main/java/org/parosproxy/paros/model/OptionsParam.java:96]: +1 for moving to L44/45. The deprecation and code changes are part of another class, would it be okay to include those here?

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/model/OptionsParam.java:96]: Not necessary.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thanks

**Pull request merged**